tests/monitoring/vm_monitoring.go

```diff
@@ -271,19 +271,27 @@ var _ = Describe("[Serial][sig-monitoring]VM Monitoring", Serial, decorators.Sig
 
 	Context("Metrics that are based on VMI connections", func() {
 		It("should have kubevirt_vmi_last_api_connection_timestamp_seconds correctly configured", func() {
-
 			By("Starting a VirtualMachineInstance")
 			vmi := libvmifact.NewAlpine()
 			vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 			vmi = libwait.WaitForSuccessfulVMIStart(vmi)
 
-			By("Validating the metric gets updated with the last connection timestamp")
+			By("Validating the metric gets updated with the first connection timestamp")
 			Expect(console.LoginToAlpine(vmi)).To(Succeed())
 			initialMetricValue := validateLastConnectionMetricValue(vmi, 0)
 
-			time.Sleep(1 * time.Minute)
+			By("Deleting the VirtualMachineInstance")
+			Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(context.Background(), vmi.Name, metav1.DeleteOptions{})).To(Succeed())
+			libwait.WaitForVirtualMachineToDisappearWithTimeout(vmi, 240)
 
+			By("Starting the same VirtualMachineInstance")
+			vmi = libvmifact.NewAlpine()
+			vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			vmi = libwait.WaitForSuccessfulVMIStart(vmi)
```

Comment:
I need to recreate the VMI in order to perform the second connection. Creating a new connection without terminating the VMI is the flaky part, sometimes the old connection session is still in place and the second isn't being made.

Reply:
Did you check with the core virt team to see if that behavior is a bug or not?

Reply:
Yes, it is expected. See
https://github.com/kubevirt/kubevirt/blob/adb4f2cb5e020678388848fd8283098adaaad889/tests/console/login.go#L92

Also see https://github.com/kubevirt/kubevirt/blob/adb4f2cb5e020678388848fd8283098adaaad889/tests/container_disk_test.go#L201-L215 where we first login, and then write commands to the console (i.e. the connection remains in place).

This test is the only test in the repo that perform two connection to the same VMI.

Reply:
So there is no process to terminate the connection? And since you want to see the metric value increasing from the first connection to the other, you need to delete the VMI to ensure that the initial connection no longer exists?

Reply:
correct